Examples/BookingSystem.AspNetCore/Feeds/OrdersFeed.cs

```diff
@@ -15,63 +16,74 @@ public class AcmeOrdersFeedRPDEGenerator : OrdersRPDEFeedModifiedTimestampAndID
                                     PriceCurrency = "GBP"
                                 },
-                                OrderedItem = thisOrder.Select(orderItem => new OrderItem
-                                {
-                                    Id = this.RenderOrderItemId(OrderType.Order, thisOrder.Key.orders.OrderId, orderItem.Id),
-                                    AcceptedOffer = new Offer { 
-                                        Id = new Uri(orderItem.OfferJsonLdId),
-                                        Price = orderItem.Price,
-                                        PriceCurrency = "GBP"
-                                    },
-                                    OrderedItem = RenderOpportunityWithOnlyId(orderItem.OpportunityJsonLdType, new Uri(orderItem.OpportunityJsonLdId)),
-                                    OrderItemStatus =
-                                        orderItem.Status == BookingStatus.Confirmed ? OrderItemStatus.OrderItemConfirmed :
-                                        orderItem.Status == BookingStatus.CustomerCancelled ? OrderItemStatus.CustomerCancelled :
-                                        orderItem.Status == BookingStatus.SellerCancelled ? OrderItemStatus.SellerCancelled :
-                                        orderItem.Status == BookingStatus.Attended ? OrderItemStatus.CustomerAttended : (OrderItemStatus?)null
+                                OrderedItem = RenderOpportunityWithOnlyId(orderItem.Item3.OpportunityJsonLdType, new Uri(orderItem.Item3.OpportunityJsonLdId)),
+                                OrderItemStatus =
+                                    orderItem.Item3.Status == BookingStatus.Confirmed ? OrderItemStatus.OrderItemConfirmed :
+                                    orderItem.Item3.Status == BookingStatus.CustomerCancelled ? OrderItemStatus.CustomerCancelled :
+                                    orderItem.Item3.Status == BookingStatus.SellerCancelled ? OrderItemStatus.SellerCancelled :
+                                    orderItem.Item3.Status == BookingStatus.Attended ? OrderItemStatus.CustomerAttended : (OrderItemStatus?)null
 
-                                }).ToList()
-                            }
-                        };
+                            }).ToList()
+                        }
+                    });
 
-            // Note there's a race condition in the in-memory database that allows records to be returned from the above query out of order when modified at the same time. The below ensures the correct order is returned.
-            var items = query.ToList().Take(this.RPDEPageSize).ToList();
+                // Note there's a race condition in the in-memory database that allows records to be returned from the above query out of order when modified at the same time. The below ensures the correct order is returned.
+                var items = query.ToList().Take(this.RPDEPageSize).ToList();
```

Comment:
We should be able to remove this hack now, as hopefully not an issue with SQLite?  Should we be using `query.Take()` direct to ensure only the `RPDEPageSize` number of rows is returned?